### DISCUSSION TITLE
General configuration question (IKEA E1743)

### DISCUSSION THREAD

**Discussion by mgv1980** [itavero/homebridge-z2m#814] [Q&A] [answered]
Hello all,
I am new in the Z2M world and desperately need your help (I spend hours looking for a solution...)

I am running Homebridge on rp4 + Zigbee2MQTT.

In Zigbee2MQTT I successfully connected my IKEA E1743 switch. Homebridge is picking up the switch from MQTT with 6 different buttons: 
![image](https://github.com/itavero/homebridge-z2m/assets/158351323/4f6c3b28-fae6-4061-8f09-49efb8a2cffb)

This buttons are not visible in the Home app on my Iphone nor are they having any reaction on any button press (except the battery charge indicator is shown correctly).

I assume that the reason is lack of configuration in the JSON file. I tried everything possible without success.

Current JSON:

{
    "mqtt": {
        "base_topic": "zigbee2mqtt",
        "server": "mqtt://192.168.178.96:1883",
        "reject_unauthorized": false,
        "keepalive": 60,
        "version": 4,
        "disable_qos": false
    },
    "defaults": {
        "exclude": false
    },
    "exclude_grouped_devices": false,
    "platform": "zigbee2mqtt"
}

For information - Homebridge is shwowing the following:

31/01/2024, 21:33:30] [zigbee2mqtt] Mapping of property 'action' of device 'ikea':
Button 1 (off):
        - SINGLE: off
Button 2 (on):
        - SINGLE: on
Button 3 (brightness_stop):
        - SINGLE: brightness_stop
Button 4 (brightness_move_down):
        - SINGLE: brightness_move_down
Button 5 (brightness_move_up):
        - SINGLE: brightness_move_up

Please point me to the correct direction.

Thank you in advance.
Marko

**Comment by itavero** [comment]
Have you tried with another app like Eve or Controller for HomeKit?

**Comment by mgv1980** [comment]
Thank you very much for your reply.

I can see the button(s) in Eve and I have assigned an action to them - however nothing happens when switch used.

**Reply by itavero** [reply]
On MQTT, do you see the same values being send as mentioned in the mapping from the Homebridge logs?

Some devices have a `legacy` setting in Zigbee2MQTT that you might have to change

**Reply by HelderCardoso** [reply]
Itavero,

Can shortcut work with this remotes?
I have not manage to do so with any remote I have.
Ikea and Tuya.

Any ideas?

Best regards

**Reply by beesMMS** [reply]
@itavero I'd be very interested in the same topic. I face the same situation than @mgv1980  and I unfortunately can't see the remote in the Apple Home App. I have changed the legacy setting as suggested by z2m so this should be fine. Do you have any suggestion how to proceed? Thanks in advance!

**Reply by itavero** [reply]
Do you see the button mapping in the Homebridge log on startup?

Have you checked with a third party app?

**Reply by beesMMS** [reply]
I guess you are talking about this mapping:
[8/14/2024, 4:41:00 PM] [zigbee2mqtt] Mapping of property 'action' of device 'WindelWechsler':
Button 1 (off):
        - SINGLE: off
Button 2 (on):
        - SINGLE: on
Button 3 (brightness_stop):
        - SINGLE: brightness_stop
Button 4 (brightness_move_down):
        - SINGLE: brightness_move_down
Button 5 (brightness_move_up):
        - SINGLE: brightness_move_up

I see it in the log of Homebridge but I cannot see the button in the Home app and I also tried the Eve which doesn't shop it neither.

**Reply by itavero** [reply]
And you do have a Home Hub configured (Apple TV, HomePod)?

I have quite some remote controls and never had such issues myself. I wouldn't really know where to start debugging this issue, which makes met think is not in the plugin, but perhaps something about the system.

![image](https://github.com/user-attachments/assets/b2da4a21-7648-4a93-ab9d-2d305e360919)

![image](https://github.com/user-attachments/assets/05f950d7-7abc-4274-bad3-cbcca36a20c3)

![image](https://github.com/user-attachments/assets/3b48fe6c-3674-47f6-aa58-dbd846aefff1)

**Reply by beesMMS** [reply]
huh.. I see :/ indeed I use an Apple TV as hub. But does this really make the difference? :/ If any other idea comes to your mind, I'd really appreciate it!

**Reply by itavero** [reply]
> indeed I use an Apple TV as hub. But does this really make the difference?

I would not expect it to. I use both an older Apple TV and HomePod Minis.
As far as I know you just need something to function as home hub.. shouldn't matter what Apple device you are using for that

Do you have the latest updates installed on it and your phone?

**Reply by beesMMS** [reply]
My Apple TV and iPhone are fully up 2 date. Only the IKEA button does not run the latest software. Do you think this makes a difference?

**Reply by beesMMS** [reply, marked as answer]
Alright... I figured the problem out.. it's quite stupid. most of the device I added to AppleHome have been shown in the Home View directly. This default behavior is different for programmable buttons! By default they are only visible when you go into the room you've added them to or to your default room.
That means, the button has been there all the time.. I just didn't look at the correct spot.

@itavero thank you anyways and sorry for wasting your time!

**Reply by beesMMS** [reply]
the only thing I din't figure out so far: can I change the mapping of actions? So far I understand, that the remote has 2 buttons and the mapped actions are single press and long press each. I was wondering.. what is the 5. action and how can I trigger it? Is it possible to change the mapping to Apple Home can also react to a double click?

**Reply by itavero** [reply]
> Is it possible to change the mapping to Apple Home can also react to a double click?

The mapping is shown on start-up of Homebridge.
It can not be changed.

```
Button 1 (off):
- SINGLE: off
Button 2 (on):
- SINGLE: on
Button 3 (brightness_stop):
- SINGLE: brightness_stop
Button 4 (brightness_move_down):
- SINGLE: brightness_move_down
Button 5 (brightness_move_up):
- SINGLE: brightness_move_up
```

The goal of homebridge-z2m is that it works "out of the box" as much a possible, without needing a lot of configuration.
If you want customizations, I typically point to a plugin like `homebridge-mqttthing`.
